crates/compilers/tests/project.rs

```diff
@@ -4154,3 +4157,33 @@ contract A { }
         );
     });
 }
+
+#[test]
+fn can_preprocess_constructors_and_creation_code() {
+    let root =
+        canonicalize(Path::new(env!("CARGO_MANIFEST_DIR")).join("../../test-data/preprocessor"))
+            .unwrap();
+
+    let paths = ProjectPathsConfig::builder()
+        .sources(root.join("src"))
+        .tests(root.join("test"))
+        .root(&root)
+        .build::<SolcLanguage>()
+        .unwrap();
+
+    let project = ProjectBuilder::<SolcCompiler>::new(Default::default())
+        .paths(paths)
+        .build(SolcCompiler::default())
+        .unwrap();
+
+    // TODO: figure out how to set root to parsing context.
+    let cur_dir = env::current_dir().unwrap();
+    env::set_current_dir(root).unwrap();
```

Comment:
https://github.com/paradigmxyz/solar/pull/231

Reply:
can pin to main now

Reply:
this todo is still up

Reply:
ah, missed the update, so basically should work as setting root as current dir in file resolver?
```rust
 parsing_context.file_resolver.set_current_dir(paths.root.clone());
```

Reply:
if doing so and removing setting the dir in test, it fails with  `file src/Counter.sol not found`